.github/workflows/tests.yml

```diff
@@ -0,0 +1,31 @@
+name: Run tests
+on: [push]
+
+env: 
+  SHOULD_RUN_COVERAGE: ${{ !(github.event.push.branches == 'master' || github.event.issue.pull_request) }}
```

Comment:
You can avoid writing environment variables by clarifying the trigger (see [here](https://futurestud.io/tutorials/github-actions-run-on-pull-request)).

Reply:
taken offline!